src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -971,7 +848,8 @@ private void FastKill()
 
             // Setting the event should be the last thing we do.
             // Do nothijng after that!
-            siloTerminatedEvent.Set();  
+            siloTerminatedTask.SetResult(0);
+            siloTerminatedEvent.Set();
```

Comment:
I believe you can just explicitly cast the siloTerminatedTask to a WaitHandle, instead of having 2 separate signals.